dcpy/utils/postgres.py

```diff
@@ -8,41 +8,163 @@
 from typing import Optional
 
 
-BUILD_ENGINE_RAW = os.environ["BUILD_ENGINE"]
-build_engine = create_engine(BUILD_ENGINE_RAW)
+def generate_engine_uri(server_url: str, database: str, schema: Optional[str] = None):
+    schemas = "public" if not schema else f"{schema},public"
+    schemas = schemas.replace("-", "_")  # no dashes in postgres schema names
+    options = f"?options=--search_path%3D{schemas}"
+    return server_url + "/" + database + options
 
 
-def execute_query(query: str, placeholders: Optional[dict] = None) -> None:
-    if placeholders is None:
-        placeholders = {}
-    with build_engine.connect() as connection:
-        connection.execute(statement=text(query), parameters=placeholders)
-
-
-def execute_select_query(
-    query: str, placeholders: Optional[dict] = None
-) -> pd.DataFrame:
-    if placeholders is None:
-        placeholders = {}
-    with build_engine.connect() as sql_conn:
-        select_records = pd.read_sql(sql=text(query), con=sql_conn, params=placeholders)
-    return select_records
-
-
-def execute_file_via_shell(build_engine: str, path: Path):
+def execute_file_via_shell(engine_uri: str, path: Path):
     """Execute .sql script at given path."""
-    cmd = f"psql {build_engine} -v ON_ERROR_STOP=1 -f {path}"
+    cmd = f"psql {engine_uri} -v ON_ERROR_STOP=1 -f {path}"
     if os.system(cmd) != 0:
         raise Exception(f"{path} has errors!")
 
 
-def execute_via_shell(build_engine: str, sql_statement):
+def execute_query_via_shell(engine_uri: str, sql_statement):
     """Execute sql via psql shell."""
-    cmd = f"psql {build_engine} -v ON_ERROR_STOP=1 {sql_statement}"
+    cmd = f"psql {engine_uri} -v ON_ERROR_STOP=1 {sql_statement}"
     if os.system(cmd) != 0:
         raise Exception(f"Command has errors! {cmd}")
 
 
+class PostgresClient:
+    def __init__(self, server_url: str, database: str, schema: str):
+        self.server_url = server_url
+        self.database = database
+        self.schema = schema.replace("-", "_")  # no dashes in postgres schema names
+        self.engine_uri = generate_engine_uri(
+            server_url=server_url, database=database, schema=schema
+        )
+        self.engine = create_engine(
+            self.engine_uri,
+            isolation_level="AUTOCOMMIT",
+        )
+        self.create_schema(self.schema)
+
+    def execute_query(self, query: str, placeholders: Optional[dict] = None) -> None:
+        if placeholders is None:
+            placeholders = {}
```

Comment:
Nit of nits, but connection.execute can just take `parameters=None` (`None` is the default value), so we don't need this